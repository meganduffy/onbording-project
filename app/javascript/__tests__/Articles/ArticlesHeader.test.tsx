import React from 'react';
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';
import ArticleHeader from '../../components/Articles/ArticlesHeader';

let container = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders correctly', ()=>{
  const tree = renderer.create(<ArticleHeader title='Article Header' subtitle='Ipsum Lorem'
   />).toJSON();
  expect(tree).toMatchSnapshot();
});

it( 'renders with a title', ()=>{
    act( ()=>{render(<ArticleHeader title='Article Header'/>, container)} )
        expect(container.textContent).toBe("Article Header")
});
