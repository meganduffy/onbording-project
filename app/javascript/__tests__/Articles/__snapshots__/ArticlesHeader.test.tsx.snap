// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesHeader Snapshots should render as expected 1`] = `
<div>
  <div
    id="articles-header"
  >
    <a
      href="http://localhost"
    >
      <h1>
        Demonic Cats
      </h1>
      <h2>
        And the people who love them.
      </h2>
    </a>
    <br />
  </div>
</div>
`;
